src/components/checkbox/CheckboxGroup.tsx

```diff
@@ -0,0 +1,92 @@
+import { ChangeEvent, useState } from 'react';
+import styled from '@emotion/styled';
+
+import Checkbox from './Checkbox';
+import CheckboxWithText from './CheckboxWithText';
+
+interface Option {
+  name: string;
+  id: string;
+  [key: string]: unknown;
+}
+
+export interface CheckboxGroupProps {
+  title: string;
+  options: Option[];
+}
+
+const CheckboxGroup = ({ title, options }: CheckboxGroupProps) => {
+  const [checkStatus, setCheckStatus] = useState<boolean[]>(new Array(options.length).fill(false));
+
+  const toggleSingleCheckbox = (clikedIdx: number) => {
+    setCheckStatus((prev) => prev.map((item, i) => (i === clikedIdx ? !item : item)));
+  };
+
+  const toggleCheckAllBtn = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.checked) {
+      setCheckStatus((prev) => [...prev.fill(true)]);
+      return;
+    }
+    setCheckStatus((prev) => [...prev.fill(false)]);
+  };
```

Comment:
```suggestion
  const toggleCheckAllBtn = (e: ChangeEvent<HTMLInputElement>) => {
    setCheckStatus((prev) => [...prev.fill(e.target.checked)]);
  };
```

축약을 하자면 이렇게도 될거 같네요 ...?! (중요하진 않아요)

Reply:
> 축약을 하자면 이렇게도 될거 같네요 ...?! (중요하진 않아요)

어머 완전 똑똑이